### PR TITLE
Fix frontend build with Angular 16 and update Sprint Boot to version 2

### PR DESCRIPTION
**Pull request by pyby**
### Motivation and Context

The frontend was too old.
It has to be replaced.

### Description

- Use Node 20.8.0 and npm 10.2.1 to build the frontend app.
- Updated application in `portal-app` folder with AngularJS 16.2.0.
- Upgrade to Sprint Boot to 2.7.17.
- Add `WebMvcConfig` to serve frontend app pages. Thanks @jboix.
- Replace custom login page with the default `loginForm` from SprintBoot. Thanks @jboix.
- Add `.editorconfig` at root to be used in both projects.
- Add Android application custom requests in code and unit tests, not in read me. 
- Unit tests are running with `mvn clean test`.

### Checklist

- [x] The branch has been rebased onto the `main` branch.
- [x] The green check mark "All checks have passed" is displayed on the PR.
- [x] The documentation has been updated (if relevant).
- [x] Issues are linked to the PR, if any.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -14,7 +14,12 @@ Playfff is a SRG micro service to serve extra datas to Play applications. Playff
 
 ## Installation
 
-A Java development environment with Maven is needed. To run the NodeJS server, `npm` and `python` (python 2) are needed as well.
+A Java 1.8 development environment with Maven is needed. To run the NodeJS server, `npm` is needed as well.
```
Comment (jboix) [inline docs/README.md:17]: The target bytecode is 1.8 but I've managed to compile and run the code with a JDK 17.

File: src/main/java/ch/srgssr/playfff/config/WebMvcConfig.java [hunk 2]
```diff
@@ -11,12 +19,27 @@
  */
 @Configuration
 public class WebMvcConfig extends WebMvcConfigurerAdapter {
+    private final static Logger log = LoggerFactory.getLogger(WebMvcConfig.class);
```
Comment (jboix) [inline src/main/java/ch/srgssr/playfff/config/WebMvcConfig.java:22]: Unused logger, can be deleted.

File: docs/README.md (hunk 1 above)
Comment (pyby) [reply docs/README.md:17]: @jboix Did you run tests as well? I'm in the middle of our CI which is working and my setup, not correctly setup.

Comment (pyby) [reply docs/README.md:17]: Good idea. Issue #78 opened. For a new PR. This one is too huge now.

**Comment by pyby** [comment]
- Unit tests run again on CI and pass.
- Play Android application custom requests are supported.
- Cache for `api/v2/deeplink/parsePlayUrl.js` is working.

**Review by jboix — APPROVED** [review]
LGTM

**Pull request merged**